components/browser/storage-sync/metrics.yaml

```diff
@@ -0,0 +1,257 @@
+      truncated and sanitized to omit PII, like URLs and file system paths.
+    send_in_pings:
+    - history_sync
+    bugs:
+    - 3092
+    data_reviews:
+    - https://github.com/mozilla-mobile/android-components/pull/3092
+    notification_emails:
+    - sync-core@mozilla.com
+    expires: never
+
+bookmarks_sync:
+  uid:
+    type: string
+    description: >
+      The user's hashed Firefox Account ID.
+    send_in_pings:
+    - bookmarks_sync
+    bugs:
+    - 3092
+    data_reviews:
+    - https://github.com/mozilla-mobile/android-components/pull/3092
+    notification_emails:
+    - sync-core@mozilla.com
+    expires: never
+  started_at:
+    type: datetime
+    time_unit: millisecond
+    description: >
+      Records when the bookmark sync started.
+    send_in_pings:
+    - bookmarks_sync
+    bugs:
+    - 3092
+    data_reviews:
+    - https://github.com/mozilla-mobile/android-components/pull/3092
+    notification_emails:
+    - sync-core@mozilla.com
+    expires: never
+  finished_at:
+    type: datetime
+    time_unit: millisecond
+    description: >
+      Records when the bookmark sync finished.
+    send_in_pings:
+    - bookmarks_sync
+    bugs:
+    - 3092
+    data_reviews:
+    - https://github.com/mozilla-mobile/android-components/pull/3092
+    notification_emails:
+    - sync-core@mozilla.com
+    expires: never
+  incoming:
+    type: labeled_counter
+    labels:
+    - applied
+    - failed_to_apply
+    - reconciled
+    description: >
+      Records incoming bookmark record counts.
+    send_in_pings:
+    - bookmarks_sync
+    bugs:
+    - 3092
+    data_reviews:
+    - https://github.com/mozilla-mobile/android-components/pull/3092
+    notification_emails:
+    - sync-core@mozilla.com
+    expires: never
+  outgoing:
+    type: labeled_counter
+    labels:
+    - uploaded
+    - failed_to_upload
+    description: >
+      Records outgoing bookmark record counts.
+    send_in_pings:
+    - bookmarks_sync
+    bugs:
+    - 3092
+    data_reviews:
+    - https://github.com/mozilla-mobile/android-components/pull/3092
+    notification_emails:
+    - sync-core@mozilla.com
+    expires: never
+  outgoing_batches:
+    type: counter
+    description: >
+      Records the number of batches needed to upload all outgoing records.
+    send_in_pings:
+    - bookmarks_sync
+    bugs:
+    - 3092
+    data_reviews:
+    - https://github.com/mozilla-mobile/android-components/pull/3092
+    notification_emails:
+    - sync-core@mozilla.com
+    expires: never
+  failure_reason:
+    type: labeled_string
+    labels:
+    - other
+    - unexpected
+    - auth
+    description: >
+      Records bookmark sync failure reasons.
+    send_in_pings:
+    - bookmarks_sync
+    bugs:
+    - 3092
+    data_reviews:
+    - https://github.com/mozilla-mobile/android-components/pull/3092
+    notification_emails:
+    - sync-core@mozilla.com
+    expires: never
+  remote_tree_problems:
```

Comment:
This is another case where recording an arbitrary timestamp via `NumberMetricType` would be nifty—in addition to recording validation problems, we'd also like to record how long validation, and each phase of the merge, took.

That's not something we can do with timespans today, because the Rust code measures that internally. Calling into Glean via the FFI from Rust definitely wouldn't be worth it for this one case.

We don't need it for the Fenix MVP; `startedAt`, `finishedAt`, and Desktop + iOS telemetry via the Sync ping is enough for now. But it's something we'd like to collect eventually. 😄

Reply:
I wonder if the timeline allows you to wait for the potential Rust API to come out? (no ETA for it yet, but I suspect somewhen 2020).

Reply:
We filed a bug about adding an API to record arbitrary timespans [here](https://bugzilla.mozilla.org/show_bug.cgi?id=1556341).

Reply:
Thanks! I think the pings we get from Desktop and iOS with the information should be enough to tide us over until the Rust Glean API is ready.

Reply:
Edit: @mdboom talked about this on Slack, and, after thinking about it, we'd really like to have a way to record raw timespans for the MVP, or soon after, so we can measure the performance of our storage layer.

Reply:
Thanks!